src/utils/BeameUtils.js

```diff
@@ -88,6 +87,77 @@ function findInTree(node, predicate, limit) {
 }
 //endregion
 
+//region Background Jobs
+/**
+ * @typedef {Object} BackgroundJob
+ * @property {Object} handle => set interval handle
+ * @property {Number} interval => interval to run the job
+ * @property {Number} called => times the job was called
+ * @property {Number} failed => times the job failed
+ * @property {Boolean} running => will be true while the process is running
+ * @property { {result: * | undefined, error: * | undefined, timestamp: Number} | undefined} lastRun => result from the last run
+ */
+/**
+ * @type {{name: BackgroundJob}}
+ */
+const backgroundJobs = {};
+/**
+ * Starts a background job that will run every interval
+ * @param {String} name
+ * @param {Function} func
+ * @param {Number} interval (in ms)
+ * @returns {boolean}
+ */
+function startBackgroundJob(name, func, interval) {
+	if(backgroundJobs[name]) {
```

Comment:
I would `assert(!backgroundJobs[name])`

Reply:
I didn't want to throw at that level. Just because the start is called 2 times with the same name we shouldn't punish the caller. Would prefer to just return false and log a warning

Reply:
done